### ISSUE TITLE
vectorGrid dependency removed>

### ISSUE THREAD

**Issue by cscrum** [bluehalo/ngx-leaflet#374]
using Angular 17, @bluehalo/ngx-leaflet 19.0.0, leaflet: 1.9.4
I recently upgraded a project from Angular 14 and am doing some major rewrites so updating all packages where I can. I have vector tiles that I am serving from my own server, but after upgrading to bluehalo from the previous maintainer, angular is complaining that:

 Module '"leaflet"' has no exported member 'vectorGrid'. 

I see in the old package.json for ngx-leaflet there was a dependency for leaflet.vectorgrid, but that is no longer included. I have tried installing leaflet.vectorgrid manually but I still get the error above. Is there a way to fix this? Or is there a better way to display vector tiles on my angular leaflet map?

I used to have:

const vectorTileOptions = {
          vectorTileLayerStyles: vectorTileStyling,
          renderer: renderer,
          zIndex: 1000,
          interactive: true, // Make sure that this VectorGrid fires mouse/pointer events
  };

const tlayer = vectorGrid.protobuf(tileurl + '/{z}/{x}/{y}.pbf',
 vectorTileOptions,
 ).on('click', function(e) { })

And I had to add a namespace in my typings.d.ts file like this:
import * as L from "leaflet";
declare module "leaflet" {
    namespace vectorGrid {
      export function protobuf(url: string, options?: any): any;
    }
}

but none of this works now. Any help would be appreciated.

**Comment by reblace** [comment]
I don't think we ever directly included Leaflet.VectorGrid in any of our releases. It's possible that something else has broken how you were including it. Can you include your old package.json?

**Comment by reblace** [comment]
Closing this out — `vectorGrid` / Leaflet.VectorGrid has never been a dependency of ngx-leaflet. It's a separate Leaflet plugin that was never bundled with or required by this library.

If you're getting `Module '"leaflet"' has no exported member 'vectorGrid'`, it's likely that a previous setup was importing VectorGrid through a different path or your old project had it as a direct dependency that didn't get carried forward in the upgrade. You'll want to install `leaflet.vectorgrid` directly and import it separately from ngx-leaflet — it should work alongside it without any changes to this library.
